packages/scaffold-config/test/unit/detect.spec.ts

```diff
@@ -304,4 +304,16 @@ describe('detectLanguage', () => {
 
     expect(actual).to.eq('ts')
   })
+
+  it('ignores node_modules when checking for tsconfig.json', async () => {
+    const projectRoot = await scaffoldMigrationProject('pristine-cjs-project')
+
+    await fs.mkdirp(path.join(projectRoot, 'node_modules', 'some-node-module'))
+    await fs.writeFile(path.join(projectRoot, 'node_modules', 'some-node-module', 'tsconfig.json'), '')
```

Comment:
Could we have stubbed these to achieve the same results instead of needing to write to the file system?'

Also are these being cleaned up? I'm prob missing it

Reply:
It's using the `scaffoldMigrationProject` function which uses the system testes `Fixtures` code - it's not cleaned up, but the files are all created inside of `/tmp`, so I don't think it's necessary to clean up - it's just doing whatever the `system-tests/fixtures` code already does.

Reply:
For a unit test I'd prefer to use something like https://github.com/tschaub/mock-fs or `sinon`, writing to the filesystem is unnecessarily slow and potentially flaky for this kind of test. Really the only time you /need/ to write to the real FS is when launching multiple processes, like in the system tests.